### PR TITLE
Fix errors that occur when WC subscriptions is activated in bulk request or updated via WC extensions screen

### LINKED ISSUE #6756: Updating Subscriptions through the WooCommerce > Extensions while WC Payments is active results in fatal error
### Describe the bug

Updating the WooCommerce Subscriptions extension via **WooCommerce > Extensions > My Subscriptions** while WooPayments is also active results in a fatal error and Subscriptions being deactivated.

image

After updating the plugin, the merchant receives the following fatal error with Subscriptions being deactivated:
image

I haven't looked into what is causing this but I think the problem here is with WooPayments loading Subscriptions Core library because it's not looking for the `action=upgrade-plugin&plugin=woocommerce-subscriptions` GET params and is loading its own subscriptions. We might also need to update `wcs_init_autoloader()` inside `woocommerce-subscriptions.php` to look for `upgrade-plugin` as well.

### To reproduce

1. Make sure WooPayments is installed and activated
2. Install the previous version of WooCommerce Subscriptions
3. Deactivate WooCommerce Subscriptions Core standalone plugin if you have it
4. Go to **WooCommerce > Extensions > My Subscriptions**
5. Scroll down to WooCommerce Subscriptions and press update
6. While trying to reactive the plugin, notice the fatal error and subscriptions being deactivated.

**Screenshots**

### Additional details

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```
</details>

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #6756

#### Changes proposed in this Pull Request

This PR addresses an issue that occurs when WC Subscriptions is updated or activated in certain scenarios.

```
PHP Fatal error:  Cannot declare class WCS_Core_Autoloader, because the name is already in use in /wp-content/plugins/woocommerce-subscriptions/vendor/woocommerce/subscriptions-core/includes/class-wcs-core-autoloader.php on line 12
```

The purpose of this PR is to update the existing code to handle additional scenarios where Subscriptions is being activated, ensuring that the error no longer occurs. The following improvements have been made:

- Previously we only checked for the `'activate'` value in the `$_GET['action']` parameter, however, when updating the plugin via **WooCommerce → Extensions** this value is `activate-plugin`, which is included in the request when updating the plugin. This modification allows us to accurately determine if Subscriptions is being activated during an update.
- Additionally, when Subscriptions is activated as part of a bulk action, the request includes the `activate-selected` value in the `$_GET['action']` parameter, along with a list of plugins being activated in the checked array. The code now takes this scenario into consideration too.

#### Updating plugin via WooCommerce → Extensions

1. Make sure WooPayments is installed and activated
2. Install the previous version of WooCommerce Subscriptions
3. Deactivate WooCommerce Subscriptions Core standalone plugin if you have it
4. Go to WooCommerce > Extensions > My Subscriptions
5. Scroll down to WooCommerce Subscriptions and press update
   - On `develop`, while trying to activate the plugin, notice the fatal error and subscriptions being deactivated.
   - On this branch there should be no error. 
   
#### Bulk activate

1. Make sure WooPayments is installed and activated
2. Go to the Plugins admin screen.
3. Deactivate WC Subscriptions.
4. Check the box next to subscriptions. 
5. Scroll to the top, select "activate" from the dropdown and run. 
   - On `develop` you should get the same fatal error mentioned above. 
   - On this brach there should be no error. 
 
#### Simple activate

> **Note**
> This wasn't previously erroring on develop. This test only exists to confirm this PR doesn't break the existing supported flow. 

1. Make sure WooPayments is installed and activated
2. Go to the Plugins admin screen.
3. Deactivate WC Subscriptions.
4. Click activate. 
5. On this branch you shouldn't get any error. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6757` or branch name `fix/error-during-subscription-update` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/error-during-subscription-update) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 75244a9669930887df2f452071a2af08f5625554
- Build time: 2023-07-14 06:09:30 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: woocommerce-payments.php [hunk 1]
```diff
@@ -171,11 +171,26 @@ function wcpay_init_subscriptions_core() {
 			$plugin_slug = "$plugin_name/$plugin_name.php";
 
 			// Check if the specified $plugin_name is in the process of being activated via the Admin > Plugins screen.
-			if ( isset( $_GET['action'], $_GET['plugin'], $_GET['_wpnonce'] ) && wp_verify_nonce( wc_clean( wp_unslash( $_GET['_wpnonce'] ) ), "activate-plugin_{$plugin_slug}" ) ) {
-				$action = sanitize_text_field( wp_unslash( $_GET['action'] ) );
-				$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
+			if ( isset( $_REQUEST['action'], $_REQUEST['_wpnonce'] ) && current_user_can( 'activate_plugin', $plugin_slug ) ) {
+				$action              = sanitize_text_field( wp_unslash( $_REQUEST['action'] ) );
+				$plugin_be_activated = '';
```
Comment (mattallan) [inline woocommerce-payments.php]: I believe `$plugin_be_activated` was supposed to be `$plugin_being_activated` 😄 
An alternative for shorter variable names could be `$activating_plugin`

**Review by mattallan — APPROVED** [review]
I've tested the following flows:

- [x] Activating/deactivating Subscriptions & Subscriptions Core using the single activate action
- [x] Bulk activate/deactivate Subscriptions & Subscriptions Core extensions at the same time
- [x] Bulk activate/deactivate Subscriptions & Subscriptions Core on their own
- [x] Upgrade WC Subscriptions via **WooCommerce -> Extensions**

File: woocommerce-payments.php (hunk 1 above)
Comment (james-allan) [reply woocommerce-payments.php]: > I believe $plugin_be_activated was supposed to be $plugin_being_activated 😄

😆 haha yes it was. I've updated the variable to `$activating_plugin` like you suggested. [ad98e60](https://github.com/Automattic/woocommerce-payments/pull/6757/commits/ad98e6090fa2b643a86e355cabae0bac12732269)

**Pull request merged**